File: libs/document-processor/src/model/channels/myos-timeline-channel.ts
import { z } from 'zod';

import { MyOSTimelineChannelSchema as RepositoryMyOSTimelineChannelSchema } from '@blue-repository/types/packages/myos/schemas/MyOSTimelineChannel';

import { channelContractBaseSchema } from '../shared/index.js';
import type { TimelineChannel } from './timeline-channel.js';

const myosTimelineChannelSchemaBase = RepositoryMyOSTimelineChannelSchema.merge(
  channelContractBaseSchema,
);

export type MyOSTimelineChannel = TimelineChannel & {
  accountId?: string;
  email?: string;
};

export const myosTimelineChannelSchema =
  myosTimelineChannelSchemaBase as z.ZodType<MyOSTimelineChannel>;

File: references/java-sdk/src/main/java/blue/language/types/myos/MyOsTimelineChannel.java
package blue.language.types.myos;

import blue.language.model.TypeBlueId;
import blue.language.types.TypeAlias;
import blue.language.types.conversation.TimelineChannel;

@TypeAlias("MyOS/MyOS Timeline Channel")
@TypeBlueId("HCF8mXnX3dFjQ8osjxb4Wzm2Nm1DoXnTYuA5sPnV7NTs")
public class MyOsTimelineChannel extends TimelineChannel {
    public String accountId;
    public String email;

    @Override
    public MyOsTimelineChannel timelineId(String timelineId) {
        super.timelineId(timelineId);
        return this;
    }

    public MyOsTimelineChannel accountId(String accountId) {
        this.accountId = accountId;
        return this;
    }

    public MyOsTimelineChannel email(String email) {
        this.email = email;
        return this;
    }
}
